lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -432,6 +437,23 @@ func (c *Client) startInputStreaming(stopCh chan struct{}) error {
 			c.cfg.Logger.WarnContext(context.Background(), "Failed reading TDP input message", "error", err)
 			return err
 		}
+		if m, ok := msg.(tdp.Ping); ok {
+			go func() {
+				// Upon receiving a ping message, we make a connection
+				// to the host and send the same message back to the proxy.
+				// The proxy will then compute the round trip time.
+				if !disableDesktopPing {
+					conn, err := net.Dial("tcp", c.cfg.Addr)
+					if err == nil {
+						conn.Close()
+					}
+				}
+				if err := c.cfg.Conn.WriteMessage(m); err != nil {
```

Comment:
This will unsynchronously write into an `io.ReadWriteCloser` - which is contractually not safe for concurrent write - that's set from a `net.Conn`, which is safe for concurrent writes from a concurrency standpoint, but with utterly meaningless results (if any of the concurrent writes is larger than one byte) since there's no telling if the underlying writes are split and in which order the bytes are going to be jumbled on the wire.

Reply:
The writer in question is `*tls.Conn` which is thread safe (internally guarded by mutex), so it wouldn't be a problem here. Could you check https://github.com/gravitational/teleport/pull/54795 where I added same guard in `tdp.Conn` so any `io.ReadWriteCloser` can work here (I simplified other parts of code, too)